clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -330,6 +330,14 @@ def warn_alias_with_section : Warning<
   "as the %select{aliasee|resolver}2">,
   InGroup<IgnoredAttributes>;
 
+let CategoryName = "Bounds Safety Issue" in {
+def err_bounds_safety_lang_not_supported : Error<
+  "bounds safety is only supported for C">;
+def warn_bounds_safety_asm_ignored : Warning<
+  "'-fbounds-safety' is ignored for assembly">,
```

Comment:
@nickdesaulniers They are separate because the languages (C++/Obj-C/etc) we want to make an error until we have a proper support vs a warning for assembly (or maybe others?) that are meant to be ignored. For the latter I think we can use `warning: argument unused during compilation` as @MaskRay suggested.

Reply:
@nickdesaulniers I managed to move -x ir test to the `.c` test. Not sure why but adding `-###` option seemed to make it work.